parler/models.py

```diff
@@ -119,6 +131,47 @@ class TranslationDoesNotExist(AttributeError, ObjectDoesNotExist):
 _lazy_verbose_name = lazy(lambda x: ugettext("{0} Translation").format(x._meta.verbose_name), six.text_type)
 
 
+def create_translations_composite_fk(shared_model, related_name, translated_model):
```

Comment:
What is this function doing and how is it related to the "select related" logic?

Reply:
This function adds 2 attributes to the main model with virtual composite foreign key to model with translations. It let us be able to join main model with translations as one-to-one twice with default and current languages. The QuerySet class adds automatically these attributes to select_related so it let have an access to translations in 2 languages reducing DB hitting.